core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/Resource.scala

```diff
@@ -64,8 +68,18 @@ object Resource extends ConfigFormat {
     def installTo(target: Path): Unit
 
     def rootEntries: List[String] = {
-      val fs = FileSystems.newFileSystem(path, this.getClass.getClassLoader)
-      Files.list(fs.getPath("/")).iterator.asScala.map(_.toString).toList
+      Try(FileSystems.newFileSystem(path, this.getClass.getClassLoader)) match {
```

Comment:
Wouldn't  be simpler with standard `try`?

```
try {
  val fs = FileSystems.newFileSystem(path, this.getClass.getClassLoader)
  Files.list(fs.getPath("/")).iterator.asScala.map(_.toString).toList
catch {
  case _: ProviderNotFoundException =>
    throw new IllegalArgumentException(
      s"""The path provided as the plugin path: $path is not valid.
         |Make sure it points to the plugin file (or .zip archive) and NOT to a directory.""".stripMargin
    )
}
```

Reply:
+1, `Try` makes sense in cases the result of a potentially-failing operation needs to be encapsulated/passed somewhere outside the method... but here the whole error handling happens inside the method, no needed for `Try`-like encapsulation